File: 0169. Majority Element/solution.py
# Boyer-Mayor Algorithm
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        curr = nums[0]
        count = 1

        for i in range(1, len(nums)):
            if curr == nums[i]: count += 1
            elif count < 1 : 
                count = 1
                curr = nums[i]
            else:
                count -= 1
        return curr

File: 0169. Majority Element/solution.c
// Boyer-Mayor Voting Algorithm
int majorityElement(int* nums, int numsSize){
    int cand = nums[0];
    int count = 1;

    for (int i = 1; i < numsSize; i++){
        if (nums[i] == cand) count++;

        else count--;

        if (count == 0){ 
        cand = nums[i];
        count = 1;
        }
    }

    return cand;
}